src/blueprints/tv2/helpers/tv2-sisyfos-persistent-layer-finder.ts

```diff
@@ -0,0 +1,75 @@
+import { Part } from '../../../model/entities/part'
+import { Piece } from '../../../model/entities/piece'
+import { Tv2PieceMetaData, Tv2SisyfosPersistenceMetaData } from '../value-objects/tv2-meta-data'
+
+export class Tv2SisyfosPersistentLayerFinder {
+  public findLayersToPersist(
+    part: Part,
+    time: number | undefined,
+    layersWantingToPersistFromPreviousPart: string[] = []
+  ): string[] {
+    if (!time) {
+      time = Date.now()
+    }
+
+    const piecesWithSisyfosMetaData: Piece[] = this.findPiecesWithSisyfosMetaData(part)
+    const lastPlayingPiece: Piece | undefined = this.findLastPlayingPiece(
+      piecesWithSisyfosMetaData,
+      part.getExecutedAt(),
+      time
+    )
+
+    // TODO: Does this work? Test it
+    if (!lastPlayingPiece) {
+      return []
+    }
+
+    // .findPieceWithSisyfosMetaData() has already filtered all Pieces without SisyfosPersistenceMetaData away, so we know it's not undefined.
+    const lastPlayingPieceMetaData: Tv2SisyfosPersistenceMetaData = (lastPlayingPiece.metaData as Tv2PieceMetaData)
+      .sisyfosPersistMetaData!
+
+    if (!lastPlayingPieceMetaData.wantsToPersistAudio) {
+      return []
+    }
+
+    if (!lastPlayingPieceMetaData.acceptsPersistedAudio) {
+      return lastPlayingPieceMetaData.sisyfosLayers
+    }
+
+    const layersToPersist: string[] = [...lastPlayingPieceMetaData.sisyfosLayers]
+    if (!lastPlayingPieceMetaData.isModifiedOrInsertedByAction) {
+      layersToPersist.push(...layersWantingToPersistFromPreviousPart)
+    } else if (lastPlayingPieceMetaData.previousSisyfosLayers) {
+      layersToPersist.push(...lastPlayingPieceMetaData.previousSisyfosLayers)
+    }
+
+    return Array.from(new Set(layersToPersist))
+  }
+
+  private findPiecesWithSisyfosMetaData(part: Part): Piece[] {
+    return part.getPieces().filter((piece) => {
+      if (!piece.metaData) {
+        return
+      }
+      const metaData: Tv2PieceMetaData = piece.metaData as Tv2PieceMetaData
```

Comment:
Also if it's the data about Meta? ;) 

It's been updated

Reply:
I had to revert some of it back to `metaData` since that's what the attribute is called in the database and it breaks the system if we name it `metadata`.
We would need to make a converter that maps it from `metaData` to `metadata`.